packages/vanilla/src/basics/form-elements/input-button/input-button.scss

```diff
@@ -45,53 +45,82 @@
   width: 16px;
   border-radius: 3px;
 
-  .hig__input-button--radio & {
-    border-radius: 50%;
+  .hig__input-button__input:hover + & {
+    border-color: color(hig-turquoise-50);
   }
 
-  .hig__input-button--radio .hig__input-button__input:checked + & {
-    box-shadow: inset 0 0 0 3px color(hig-white);
+  .hig__input-button--checkbox .hig__input-button__input:checked + &  {
+    background: no-repeat  color(hig-slate-40) url('../../icons/src/check-white-24.svg') center;
+    border-color: color(hig-slate-40);
   }
 
- 
-  .hig__input-button:hover & {
-    border-color: color(hig-turquoise-50);
+  .hig__input-button--checkbox .hig__input-button__input:focus:checked + & {
+    background: no-repeat   color(hig-blue-60) url('../../icons/src/check-white-24.svg') center;
+    border-color: color(hig-blue-60);
+  }
+
+
+  .hig__input-button--checkbox.hig__input-button--indeterminate .hig__input-button__input:focus + & {
```

Comment:
Consider using `:indeterminate` rather than a class for styling the indeterminate state.